eegio/base/objects/bids/bids_patient.py

```diff
@@ -0,0 +1,380 @@
+
+            # overwrite default line frequency noise in MNEBids
+            if line_noise != 50:
+                sidecardict = self.loader.load_sidecar_json()
+                sidecardict["PowerLineFrequency"] = line_noise
+                self.writer.write_sidecar_json()
+
+            output_fpaths.append(bids_edf_fpath)
+
+        # process these all into fif files
+        return output_fpaths
+
+    def load_dataset(self, session_id, run_id, kind):
+        """
+        Loads in MNE.io.Raw dataset based on session id and run id passed in. Also returns a metadata structure
+
+        Parameters
+        ----------
+        session_id :
+        run_id :
+
+        Returns
+        -------
+
+        """
+        dataset = self.loader.load_dataset(session_id, run_id)
+        metadata = self.loader.load_sidecar_json()
+
+    def _add_resected_contacts(self):
+        resected_list = " ".join(self.resected_contacts)
+
+        participants_data = self.loader.load_participants_tsv()
+        participants_json = self.loader.load_participants_json()
+        if 'resected_contacts' not in participants_json.keys():
+            participants_json = BidsBuilder.add_participant_field('resected_contacts', [], ['The list of contacts corresponding to '
+                                                                        'resected regions'])
+            self.writer.write_participants_json(participants_json)
+        colnames = participants_data.columns
+        if 'resected_contacts' not in colnames:
+            resected = [''] * participants_data.shape[0]
+            participants_data['resected_contacts'] = resected
+        for ind, row in participants_data.iterrows():
+            if row['participant_id'] == self.subject_id:
+                participants_data['paricipant_id'][ind] = resected_list
+        self.writer.write_participants_tsv(participants_data)
+
+    def _add_original_patient_name(self):
+        participants_data = self.loader.load_participants_tsv()
+        participants_json = self.loader.load_participants_json()
+        if 'original_subjectid' not in participants_json.keys():
+            participants_json = BidsBuilder.add_participant_field(participants_json, 'original_subjectid', [],
+                                              ['The subjectid in the clinical data source'])
+            self.writer.write_participants_json(participants_json)
+        colnames = participants_data.columns
+        if 'original_subjectid' not in colnames:
+            originals = [''] * participants_data.shape[0]
+            participants_data['original'] = originals
+        for ind, row in participants_data.iterrows():
+            if row['participant_id'] == self.subject_id:
+                participants_data['participant_id'][ind] = self.original_subject_id
+        self.writer.write_participants_tsv(participants_data)
+        # TODO: Add the self.original_subject_id field into the DataFrame and write to tsv
+
```

Comment:
I think these all could be the same function? i.e. it is all 

- adding a data field that is associated with a participant stored in the participants tsv, So you should be just passing in field_name=str, and col_name=str. So basically anyone can add additional data however they specify, whether it is resected, or ablated contacts, or whatever they define.

Reply:
@pmyers16